File: src/testharness/session-result-test-builder.ts
import type { SessionResult } from "../domain/value-objects/session-result.js";

export class SessionResultTestBuilder {
  private output = "default output";
  private sessionId = "sess-default";

  private constructor() {}

  static create(): SessionResultTestBuilder {
    return new SessionResultTestBuilder();
  }

  withOutput(output: string): SessionResultTestBuilder {
    this.output = output;
    return this;
  }

  withSessionId(sessionId: string): SessionResultTestBuilder {
    this.sessionId = sessionId;
    return this;
  }

  build(): SessionResult {
    return { output: this.output, sessionId: this.sessionId };
  }
}

File: src_csharp/haas/HaaS.Domain.Tests/Builders/SessionResultTestBuilder.cs
using HaaS.Domain.ValueObjects;

namespace HaaS.Domain.Tests.Builders;

public class SessionResultTestBuilder
{
    private string _output = "default output";
    private string _sessionId = "sess-default";

    private SessionResultTestBuilder() { }

    public static SessionResultTestBuilder Create() => new();

    public SessionResultTestBuilder WithOutput(string output)
    {
        _output = output;
        return this;
    }

    public SessionResultTestBuilder WithSessionId(string sessionId)
    {
        _sessionId = sessionId;
        return this;
    }

    public SessionResult Build() => new(_output, _sessionId);
}
